addons/website_project/controllers/main.py

```diff
@@ -30,9 +30,19 @@ def extract_data(self, model, values):
         data = super().extract_data(model, values)
         if model.model == 'project.task' and values.get('email_from'):
             partners_list = request.env['mail.thread'].sudo()._mail_find_partner_from_emails([values['email_from']])
-            partner_id = partners_list[0].id if partners_list else False
-            data['record']['partner_id'] = partner_id
-            data['record']['email_from'] = values['email_from']
-            if not partner_id:
+            partner = partners_list[0] if partners_list else self.env['res.partner']
+            if partner:
+                data['record']['partner_id'] = partner.id
+                data['record']['email_from'] = values['email_from']
```

Comment:
Actually, before your changes `email_from` was given even if the partner is not found. 🤔 

Is it expected to only set it when we find a partner?

Reply:
> Actually, before your changes `email_from` was given even if the partner is not found. 🤔
> 
> Is it expected to only set it when we find a partner?

email_from is necessary for both partner and non-partner. I have made the changes.

FYI- We have to pass the  partner_id; otherwise, the task will have an empty customer field

thanks